### PR TITLE
Simplify function call creation

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add the magic method `__call__` to `FunctionDef`, `Interface` and `PyccelFunctionDef` to create `FunctionCall` nodes. This makes the code more readable as the verbose `FunctionCall(func, [arg1, arg2])` can now be written as `func(arg1, arg2)`.

### CODE REVIEW

File: pyccel/ast/core.py [hunk 1]
```diff
@@ -2955,6 +2960,9 @@ def __getnewargs__(self):
             'argument_description':self._argument_description}
         return args, kwargs
 
+    def __call__(self, *args, **kwargs):
+        return self._cls_name(*args, **kwargs)
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/core.py:2965]: /bot accept
It is important to include this case in the subclass to avoid misuse. I think this line will be used in #2016

**Review by jalalium — APPROVED** [review]
Very good addition!

**Pull request merged**